src/Symfony/Component/Security/Http/Firewall/RememberMeListener.php

```diff
@@ -68,7 +68,25 @@ public function handle(GetResponseEvent $event)
         }
 
         $request = $event->getRequest();
-        if (null === $token = $this->rememberMeServices->autoLogin($request)) {
+        try {
+            if (null === $token = $this->rememberMeServices->autoLogin($request)) {
+                return;
+            }
+        } catch (AuthenticationException $e) {
+            if (null !== $this->logger) {
+                $this->logger->warning(
+                    'The token storage was not populated with remember-me token as the'
+                   .' RememberMeServices was not able to create a token from the remember'
+                   .' me information.', array('exception' => $e)
+                );
+            }
+
+            $this->rememberMeServices->loginFail($request);
+
+            if (!$this->catchExceptions) {
+                throw $e;
+            }
+
             return;
         }
```

Comment:
Look at the rest of this method for full context. The new code basically tries to treat a failure from `autoLogin` just like the failure below. I only didn't move the call into the `try-catch`, so that we could have slightly different log messages.